### ISSUE TITLE
bn_feature_net always saves last epoch?

### ISSUE THREAD

**Issue by ngreenwald** [vanvalenlab/deepcell-tf#290]
https://github.com/vanvalenlab/deepcell-tf/blob/98ea433a2f66181559d404ef92bee727a581ba2c/deepcell/training.py#L445

This could definitely be a misunderstanding on my end, but it appears that the current training.py script for bn_feature_nets always saves the latest configuration of the model, not the one with the best val_loss. 

The line referenced above occurs after all of the model callbacks, right before exiting the function. I verified that even after 4 epochs with no decrease in val loss, the .h5 file is updated after the final epoch. It could be that this is just saving the same, val_loss optimized weights again, rather than the weights from the final epoch, but if so what's the point of this line?

I deleted it in #289 , since that's when I noticed the issue. If this is not what's actually going on happy to put the line back.

**Comment by willgraf** [comment]
This is a good point.  We likely should not overwrite the final epoch always. Simply deleting the line seems like a simple fix for this.  I see no reason to save the model again inside the training function.  If we want that functionality we can just code it into our example notebooks.

I personally want to move away from our "do everything" training functions as it makes it quite difficult to customize any portion of the training flow.
